Microsoft.Toolkit.Uwp.UI.Controls/WrapPanel/WrapPanel.cs

```diff
@@ -179,42 +199,59 @@ protected override Size MeasureOverride(Size availableSize)
         /// <inheritdoc />
         protected override Size ArrangeOverride(Size finalSize)
         {
-            var parentMeasure = new UvMeasure(Orientation, finalSize.Width, finalSize.Height);
-            var spacingMeasure = new UvMeasure(Orientation, HorizontalSpacing, VerticalSpacing);
-            var paddingStart = new UvMeasure(Orientation, Padding.Left, Padding.Top);
-            var paddingEnd = new UvMeasure(Orientation, Padding.Right, Padding.Bottom);
-            var position = new UvMeasure(Orientation, Padding.Left, Padding.Top);
-
-            double currentV = 0;
-            foreach (var child in Children)
+            if (Children.Count > 0)
             {
-                var desiredMeasure = new UvMeasure(Orientation, child.DesiredSize.Width, child.DesiredSize.Height);
-                if (desiredMeasure.U == 0)
-                {
-                    continue; // if an item is collapsed, avoid adding the spacing
-                }
+                var parentMeasure = new UvMeasure(Orientation, finalSize.Width, finalSize.Height);
+                var spacingMeasure = new UvMeasure(Orientation, HorizontalSpacing, VerticalSpacing);
+                var paddingStart = new UvMeasure(Orientation, Padding.Left, Padding.Top);
+                var paddingEnd = new UvMeasure(Orientation, Padding.Right, Padding.Bottom);
+                var position = new UvMeasure(Orientation, Padding.Left, Padding.Top);
 
-                if ((desiredMeasure.U + position.U + paddingEnd.U) > parentMeasure.U)
+                double currentV = 0;
+                void arrange(UIElement child, bool isLast = false)
                 {
-                    // next row!
-                    position.U = paddingStart.U;
-                    position.V += currentV + spacingMeasure.V;
-                    currentV = 0;
-                }
+                    var desiredMeasure = new UvMeasure(Orientation, child.DesiredSize.Width, child.DesiredSize.Height);
+                    if (desiredMeasure.U == 0)
+                    {
+                        return; // if an item is collapsed, avoid adding the spacing
```

Comment:
Just thinking, if the last item in the Children is collapsed, then the one that's visible won't get stretched eh?

Reply:
😐 This is a slippery slope... What if all/multiple are collapsed? Should I consider if the child has a 0,0 size? That requires that I monitor child visibility/size while iterating to maintain the correct "viable" child for stretching... I vote no.

Reply:
Yeah, I think we need a better story around a Panel Base class helper which handles these types of questions along with Drag and Drop and all the like.  We should just call it out in the documentation for now I guess, as I can see this coming up somewhere in the future...